### PR TITLE
Nicer behavior for Option[Producer] dependencies

### PR DESCRIPTION
**Pull request by jakemannix**
- compute signature so providing Some(p) and p get the same results, and
- signature of a producer with None is the same as not having that arg at all.

### CODE REVIEW

**Comment by rodneykinney** [comment]
LGTM.  Assigning back.

File: src/main/scala/org/allenai/pipeline/Signature.scala [hunk 1]
```diff
@@ -60,20 +60,24 @@ object Signature {
...
+    val pipelineRunners = collection.mutable.ListBuffer[(String, PipelineRunnerSupport)]()
+    val otherParams = collection.mutable.ListBuffer[(String, Any)]()
+    // parameters of the following types get treated as "dependencies":
+    // PipelineRunnerSupport, Iterable[PipelineRunnerSupport], Option[PipelineRunnerSupport]
+    params.foreach {
+      case (id, p: PipelineRunnerSupport) => pipelineRunners += ((id, p))
+      case (id, it: Iterable[PipelineRunnerSupport]) =>
+        pipelineRunners ++= it.zipWithIndex.map { case (p, i) => (s"${id}_$i", p)}
+      case (id, opt: Some[PipelineRunnerSupport]) =>
+        pipelineRunners += ((id, opt.get))
```
Comment (schmmd) [inline src/main/scala/org/allenai/pipeline/Signature.scala:72]: Unnecessary extra `()`?  I think you can just `(id, opt.get)`.

Comment (jakemannix) [inline src/main/scala/org/allenai/pipeline/Signature.scala:72]: IntelliJ complained?

**Comment by rodneykinney** [comment]
This was merged in a different PR.

**Pull request closed without merging**